### ISSUE TITLE
Error output from rustdoc is stripped of color

### ISSUE THREAD

**Issue by jyn514** [obi1kenobi/cargo-semver-checks#326]
### Steps to reproduce the bug with the above code

1. Create any cargo project (e.g. `cargo new --lib example`)
2. Modify it so it doesn't compile (e.g. `echo x > src/lib.rs`)
3. Run `cargo semver-checks check-release`

### Actual Behaviour

The errors are printed without color:

<img width="1458" alt="Screenshot 2023-01-25 at 11 15 04 AM" src="https://user-images.githubusercontent.com/23638587/214633761-f32d336f-eedc-45df-a650-09bbb847993f.png">

### Expected Behaviour

The errors are printed with color, just like `cargo doc`:

<img width="1449" alt="Screenshot 2023-01-25 at 11 15 47 AM" src="https://user-images.githubusercontent.com/23638587/214634072-de81e6ee-cb12-4d90-a39d-1faf4afb7f4d.png">

#### Software version

cargo-semver-checks 0.16.2

#### Operating system

macOS 13.1 (Darwin 22.2.0)

#### Command-line

```bash
/Users/jyn/.local/lib/cargo/bin/cargo-semver-checks semver-checks --bugreport 
```

#### cargo version

```
> cargo -V 
cargo 1.68.0-nightly (a5d47a725 2023-01-16)
```

#### Compile time information

- Profile: release
- Target triple: aarch64-apple-darwin
- Family: unix
- OS: macos
- Architecture: aarch64
- Pointer width: 64
- Endian: little
- CPU features: aes,crc,dit,dotprod,dpb,dpb2,fcma,fhm,flagm,fp16,frintts,jsconv,lor,lse,neon,paca,pacg,pan,pmuv3,ras,rcpc,rcpc2,rdm,sb,sha2,sha3,ssbs,vh
- Host: aarch64-apple-darwin

### Build Configuration

```
; env | grep CARGO_
CARGO_HOME=/Users/jyn/.local/lib/cargo
CARGO_TARGET_DIR=/Users/jyn/.local/lib/cargo/target
```

### Additional Context

I see that you're passing `--json=diagnostic-rendered-ansi`, so I'm not sure where the color stripping happens. Maybe cargo doesn't think it's being output to a terminal and you need to pass `--color` explicitly?

**Comment by jyn514** [comment]
Interestingly, this works fine when I pass `-v`, it's only the default output that has color stripped.

**Comment by obi1kenobi** [comment]
Thanks for the two bug reports, much appreciated! I also suspect "cargo doesn't think it's printing to a terminal" as the cause. Will look into it!

**Comment by tonowak** [comment]
I've looked into it a bit more:

Depending on whether we want silenced output or not (so whether we use e.g. `--verbose` or not), we capture the output (by pipe'ing) or we don't: https://github.com/obi1kenobi/cargo-semver-checks/blob/135cb331a045ae20b249401b9acd2f026a142f68/src/rustdoc_cmd.rs#L57. When we don't capture, everything is fine. Otherwise, `cargo doc` sees that it's being piped and disables colors. An easy solution would be to pass `--color=always` to the `cargo doc` command (I've checked it and it works), or we could somehow try to capture the output in some other ways.

Is is OK to use `--color=always`? Honestly I don't even where to look for a terminal emulator without colors, to test whether it would work on machines without color.

**Comment by jyn514** [comment]
You can use `atty::is(atty::Stream::Stdout)` to detect if you're outputting to a terminal or not.

**Comment by obi1kenobi** [comment]
This will be out in the next release, ETA ~half an hour or so.
